apps/mobile-wallet/src/features/nftsDisplay/NftGridModal.tsx

```diff
@@ -19,23 +19,18 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 import { NFT } from '@alephium/shared'
 
 import NFTsGrid from '~/components/NFTsGrid'
-import BottomModal from '~/features/modals/BottomModal'
-import { ModalContent } from '~/features/modals/ModalContent'
+import BottomModalFlashList from '~/features/modals/BottomModalFlashList'
 import withModal from '~/features/modals/withModal'
 
 interface NftGridModalProps {
-  nftsData: NFT[]
+  addressHash?: string
+  nftsData?: NFT[]
 }
 
-const NftGridModal = withModal<NftGridModalProps>(({ id, nftsData }) => (
-  <BottomModal
-    id={id}
-    Content={(props) => (
-      <ModalContent {...props}>
-        <NFTsGrid nfts={nftsData} scrollEnabled={false} />
-      </ModalContent>
-    )}
-  />
+const NftGridModal = withModal<NftGridModalProps>(({ id, addressHash, nftsData }) => (
+  <BottomModalFlashList modalId={id} noPadding>
+    {(props) => <NFTsGrid addressHash={addressHash} nfts={nftsData} {...props} />}
```

Comment:
very very weird syntax, I would never expect the `children` prop to be a function, it took me a few minutes to figure out what's going on here.

Reply:
It's a pattern that I've seen in other libraries in the past. Can't remember which ones, but I could dig.
Would you prefer going back to having a `FlashList` prop in `BottomModalFlashList`?

Reply:
I read the blogs, left a comment in the other thread